### PR TITLE
new tools to get os paths, snap names and cleanup the snaps

### PR DESCRIPTION
**Pull request by sergiocazzolato**
The change include 3 new tools and their tests

The tools are:
os.paths: get paths information for the current system
snaps.name: get the name of a specific snap for the current system
snaps.cleanup: cleanup the snaps installed in the current system

### CODE REVIEW

File: tools/snaps.cleanup [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+    # remove all app snaps first
+    for snap in "$snap_mount_dir"/*; do
+        snap="${snap:6}"
+        case "$snap" in
+            "bin" | "$gadget_name" | "$kernel_name" | "$core_name" | "snapd" |README)
+                ;;
+            *)
+                # Check if a snap should be kept, there's a list of those in spread.yaml.
+                keep=0
```
Comment (Meulengracht) [inline tools/snaps.cleanup]: should keep (and others in this function) be marked local?

File: tools/snaps.name [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+gadget_name() {
+    echo "$(snap list | grep 'gadget$' | awk '{ print $1 }')"
+}
+
+kernel_name(){
+    echo "$(snap list | grep 'kernel$' | awk '{ print $1 }')"
+}
+
+core_name(){
+    core_name="$(snap model --verbose | grep -Po "^base:\\s+\\K.*" || true)"
```
Comment (Meulengracht) [inline tools/snaps.name:19]: local core_name?

**Review by Meulengracht — APPROVED** [review]
Looks good, just two questions

File: tools/snaps.cleanup (hunk 1 above)
Comment (sergiocazzolato) [reply tools/snaps.cleanup]: done

**Pull request merged**